packages/react/src/migrations/update-8-3-0/update-8-3-0.ts

```diff
@@ -0,0 +1,149 @@
+} from '@angular-devkit/schematics';
+import { stripIndents } from '@angular-devkit/core/src/utils/literals';
+import {
+  updatePackageJsonDependencies,
+  formatFiles,
+  insert,
+  updateJsonInTree,
+  readJsonInTree
+} from '@nrwl/workspace';
+import {
+  createSourceFile,
+  isImportDeclaration,
+  isStringLiteral,
+  ScriptTarget
+} from 'typescript';
+import { ReplaceChange } from '@nrwl/workspace/src/utils/ast-utils';
+import { relative } from 'path';
+import {
+  testingLibraryVersion,
+  frameworkVersion,
+  domTypesVersion,
+  styledComponentsTypesVersion,
+  emotionVersion,
+  reactRouterVersion,
+  styledComponentsVersion,
+  typesVersion
+} from '../../utils/versions';
+
+const ignore = require('ignore');
+
+export default function update(): Rule {
+  return chain([
+    displayInformation,
+    updateDependencies,
+    updateImports,
+    formatFiles()
+  ]);
+}
+
+function displayInformation(host: Tree, context: SchematicContext) {
+  context.logger.info(stripIndents`
+    React Testing Library has been repackaged as \`@testing-library/react\` as of version 8.
+    
+    We are replacing your \`react-testing-library\` imports with \`@testing-library/react\`.
+    
+    See: https://github.com/testing-library/react-testing-library/releases/tag/v8.0.0
+  `);
+}
+
+function updateDependencies(tree: Tree) {
+  const removeOld = updateJsonInTree(
+    'package.json',
+    (json, context: SchematicContext) => {
+      json.dependencies = json.dependencies || {};
+      json.devDependencies = json.devDependencies || {};
+      // Just in case user installed it to dependencies instead of devDependencies.
+      delete json.dependencies['react-testing-library'];
+      delete json.devDependencies['react-testing-library'];
+      context.logger.info(`Removing \`react-testing-library\` as a dependency`);
+      return json;
+    }
+  );
+
+  const packageJson = readJsonInTree(tree, '/package.json');
+
+  const candidates = {
+    react: frameworkVersion,
+    'react-dom': frameworkVersion,
+    'react-router-dom': reactRouterVersion,
+    '@types/react': typesVersion,
+    '@types/react-dom': domTypesVersion,
+    'styled-components': styledComponentsVersion,
+    '@types/styled-components': styledComponentsTypesVersion,
+    '@emotion/styled': emotionVersion
+  };
+
+  const updated = Object.entries(candidates).reduce(
+    (acc, [m, v]) => {
+      if (packageJson.dependencies[m]) {
+        acc.dependencies[m] = v;
+      } else if (packageJson.devDependencies[m]) {
+        acc.devDependencies[m] = v;
+      }
+      return acc;
+    },
+    {
+      dependencies: {},
+      devDependencies: {
+        '@testing-library/react': testingLibraryVersion
+      }
+    }
+  );
+
+  const addNew = updatePackageJsonDependencies(
+    updated.dependencies,
+    updated.devDependencies
+  );
+
+  return chain([removeOld, addNew]);
+}
+
+function updateImports(host: Tree) {
+  let ig = ignore().add(['*', '!*.ts']); // include only .ts files
```

Comment:
While this is fancy isn't it faster to just early bail here?

Also, does this consider tsx?

Reply:
Updated with `!*.tsx` as well. I don't think speed really matters here, it's gonna be less than a millisecond difference.